tests/slice_by_condition_test.py

```diff
@@ -84,15 +84,75 @@ def test_non_boolean_condition_raises_DTypeError():
         var[condition]
 
 
+def test_all_false_condition_with_wrong_dims_raises_DimensionError():
```

Comment:
I know it shouldn't be DRY, but you could parameterize the test over different values for the condition. This might better highlight that the wrong_dims tests are missing one of the cases from the wrong_shape tests.

Reply:
Ok.
A bit late now, but you should still add a case with mixed true/false.

Reply:
I'll open a new PR.